omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,300 @@
+
+        if (identifierSourceType == null || StringUtils.isBlank(identifierSourceType.toString())) {
+            errors.add("source type"); 
+        }
+        if (name == null || StringUtils.isBlank(name.toString())) {
+            errors.add("name"); 
+        }
+        if (identifierTypeUuid == null || StringUtils.isBlank(identifierTypeUuid.toString())) {
+            errors.add("patient identifier type"); 
+        }else{
+            patientIdentifierType = Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid.toString());
+            if(patientIdentifierType == null){
+                errors.add("patient identifier type");
+            }
+        }
+        if(errors.toString().replaceAll("[\\[\\]]", "").length() > 0) {
+            throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+        }
+
+        if(identifierSourceType.equals("SequentialIdentifierGenerator")) {
+            SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+            Object firstIdentifierBase = postBody.get("firstIdentifierBase");
+            Object baseCharacterSet = postBody.get("baseCharacterSet");
+            Object prefix = postBody.get("prefix");
+            Object suffix = postBody.get("suffix");
+            Object minLength = postBody.get("minLength");
+            Object maxLength = postBody.get("maxLength");
+            
+            if (firstIdentifierBase == null || StringUtils.isBlank(firstIdentifierBase.toString())) { 
+                errors.add("first identifier base"); 
+            }
+            if (baseCharacterSet == null || StringUtils.isBlank(baseCharacterSet.toString())) { 
+                errors.add("base character set"); 
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (prefix != null && StringUtils.isNotBlank(prefix.toString())) { 
+                identifierSource.setPrefix(prefix.toString()); 
+            }
+            if (suffix != null && StringUtils.isNotBlank(suffix.toString())) { 
+                identifierSource.setSuffix(suffix.toString()); 
+            }
+            if (minLength != null && StringUtils.isNotBlank(minLength.toString())) {
+                if(StringUtils.isNumeric(minLength.toString())) {
+                        identifierSource.setMinLength(Integer.parseInt(minLength.toString()));
+                }
+            }
+            if (maxLength != null && StringUtils.isNotBlank(maxLength.toString())) {
+                if(StringUtils.isNumeric(maxLength.toString())) {
+                        identifierSource.setMaxLength(Integer.parseInt(maxLength.toString()));
+                }
+            }
+            if(errors.toString().replaceAll("[\\[\\]]", "").length() > 0) {
```

Comment:
Am struggling to understand why you need to first call toString() is all you want to know is how many items are in this list

Reply:
replaced with: `if(errors.size() > 0) {}`